zsign.cpp

```diff
@@ -64,6 +65,30 @@ int usage()
 	return -1;
 }
 
+string getCacheDirectory() {
+    try {
+#ifdef _WIN32
+        const char* buffer = std::GetEnvironmentVariable("USERPROFILE");
+        if (!buffer) {
+            throw std::runtime_error("USERPROFILE environment variable is not set.");
+        }
+#else
+        const char* buffer = std::getenv("HOME");
+        if (!buffer) {
+            throw std::runtime_error("HOME environment variable is not set.");
+        }
+#endif
+        string cacheDir = std::filesystem::path(buffer) / ".cache";
+        if (!std::filesystem::exists(cacheDir) || !std::filesystem::is_directory(cacheDir)) {
```

Comment:
The directory would have to be created in case it does not exist.

The path formed in `cacheDir` most likely exists on a GNU/Linux system (not expected to be there on Win32 though).